src/plugins/console/server/routes/api/console/proxy/create_handler.ts

```diff
@@ -127,15 +129,22 @@ export const createHandler = ({
       },
     });
   } catch (e: any) {
-    log.error(e);
     const isResponseErrorFlag = isResponseError(e);
+
+    const errorMessage = isResponseErrorFlag ? JSON.stringify(e.meta.body) : e.message;
+    // core http route handler has special logic that asks for stream readable input to pass error opaquely
+    const errorResponseBody = new Readable({
+      read() {
+        this.push(errorMessage);
+        this.push(null);
```

Comment:
What's the purpose of push null?

Reply:
In a stream.Readable, calling push(null) signals the end of the stream. It tells the consumer of the stream that there is no more data to read, and that the stream has ended.